src/qt/rpcconsole.cpp

```diff
@@ -301,6 +302,14 @@ bool RPCConsole::RPCParseCommandLine(std::string &strResult, const std::string &
                             JSONRPCRequest req;
                             req.params = RPCConvertValues(stack.back()[0], std::vector<std::string>(stack.back().begin() + 1, stack.back().end()));
                             req.strMethod = stack.back()[0];
+#ifdef ENABLE_WALLET
+                            // TODO: Move this logic to WalletModel
+                            if (vpwallets.size() > 1)
+                            {
+                                // in Qt, use always the wallet with index 0 when running with multiple wallets
+                                req.URI = "/wallet/"+QUrl(QString::fromStdString(vpwallets[0]->GetName())).toString().toStdString();
```

Comment:
What about
```c++
QByteArray encodedName = QUrl::toPercentEncoding(QString::fromStdString(vpwallets[0]->GetName())));
req.URI = "wallet/"+std::string(encodedName.constData(), encodedName.length());
```
Still fairly ugly, but this makes it clear that the point is to URL-encode (percent encode) the wallet name. Not convinced that passing a string into `QUrl()` does the right thing in all cases.